app/src/main/java/com/example/scoutconcordia/Graph.java

```diff
@@ -0,0 +1,176 @@
+        return 1;
+    }
+
+    // returns true if adjacent, false if not
+    public boolean areAdjacent(LatLng ele1, LatLng ele2)
+    {
+        int id1 = getID(ele1), id2 = getID(ele2);
+        if (id1 < 0 || id2 < 0)
+            return false;
+        if (nodes[id1].adjacencyList.contains(nodes[id2]))
+                return true;
+        return false;
+    }
+    
+    public boolean replace(LatLng oldCoordinate, LatLng newCoordinate)
+    {
+        int id = getID(oldCoordinate);
+        if (id < 0)
+            return false;
+        nodes[id].element = newCoordinate;
+        return true;
+    }
+    
+    // returns -1 if the elements can't be found in the nodes of the graph, 0 if one of the elements couldn't be removed / wasn't in the adjacency list and 1 if it was successful
+    public int removeEdge(LatLng ele1, LatLng ele2)
+    {
+        int id1 = getID(ele1), id2 = getID(ele2);
+        if (id1 < 0 || id2 < 0)
+            return -1;
+        // get the indexes of the obj's in the linked list
+        int indexOfEle1inListofEle2 = nodes[id2].adjacencyList.indexOf(nodes[id1]);
+        int indexOfEle2inListofEle1 = nodes[id1].adjacencyList.indexOf(nodes[id2]);
+        if (indexOfEle1inListofEle2 < 0 || indexOfEle2inListofEle1 < 0)
+            return 0;
+        nodes[id1].adjacencyList.removeOBJ(nodes[id2]);
+        nodes[id2].adjacencyList.removeOBJ(nodes[id1]);
+        return 1;
+    }
+    
+    // returns -1 if element isn't in the graph, 0 if the element to be removed wasn't in one of it's own adjacent entries / system error and 1 if successful
+    public int removeVertex(LatLng removeMe)
+    {
+        int id = getID(removeMe);
+        if (id < 0)
+            return -1;
+        Object[] adjacentVerticies = nodes[id].adjacencyList.toArray();
+        for(int i = 0; i < adjacentVerticies.length; i++)
+        {
+            if (removeEdge(nodes[id].element, ((Node)adjacentVerticies[i]).element) != 1)
+                return 0;
+        }
+        nodes[id] = null;
+        numberOfNodes--;
+        return 1;
+    }
+    
+    public LatLng[] incidentVerticies(LatLng coordinate)
+    {
+        int id = getID(coordinate);
+        Object[] gotten = nodes[id].adjacencyList.toArray();
+        LatLng[] returnMe = new LatLng[gotten.length];
+        for (int i = 0; i < gotten.length; i++)
+        {
+            returnMe[i] = ((Node)gotten[i]).element;
+        }
+        return returnMe;
+    }
+
+    // returns an array of all the elements in the graph
+    public LatLng[] vertices()
+    {
+        int getMe = -1, placeMe = 0;
+        LatLng[] returnMe = new LatLng[numberOfNodes];
+        for (int i = 0; i < nodes.length; i++)
+        {
+            if (nodes[i] != null)
+            {
+                returnMe[placeMe] = new LatLng(nodes[i].getElement().latitude, nodes[i].getElement().longitude);
+                placeMe++;
+            }
+        }
+        return returnMe;
+    }
```

Comment:
What does a LatLng[] array look like as a String output?

How would one write a unit test for this method?

Reply:
@jaychenblue 
> 
> 
> What does a LatLng[] array look like as a String output?


As with any array you would need to cycle through it to obtain visible results otherwise an array sent out directly to the toString() method returns simply the address of the initial value. Another method would be to use Java's Arrays method to receive actual output, this is described in the following issue:
https://stackoverflow.com/questions/409784/whats-the-simplest-way-to-print-a-java-array

What the output wud be like would depend on either method used, in any case you would receive google's LatLng objects: https://developers.google.com/android/reference/com/google/android/gms/maps/model/LatLng

They display within parenthesis both of their fields / variables / attributes latitude and longitude followed in that order. Both are represented as doubles thus u can expect a number followed by several digits after the decimal point.

> 
> How would one write a unit test for this method?

This would first be conducted by having the appropriate tests verify that prior to this one, the graph and it's state were working and verified. From this point we can create a graph, insert a number of points within it and assert that every value within the array afterwards corresponds to the same point. For comparison the equals method within google's LatLng object can be used.

Reply:
Sounds good. I'll look into the links you provided when I start writing unit tests for this class.